providers/joker/auditrecords.go

```diff
@@ -0,0 +1,71 @@
+package joker
+
+import (
+	"fmt"
+
+	"github.com/StackExchange/dnscontrol/v4/models"
+)
+
+// AuditRecords returns a list of errors corresponding to the records
+// that aren't supported by this provider. If all records are
+// supported, an empty list is returned.
+func AuditRecords(records []*models.RecordConfig) []error {
+	var errs []error
+
+	// Supported record types
+	supported := map[string]bool{
+		"A":     true,
+		"AAAA":  true,
+		"CAA":   true,
+		"CNAME": true,
+		"MX":    true,
+		"NAPTR": true,
+		"NS":    true,
+		"SRV":   true,
+		"TXT":   true,
+	}
+
+	for _, rc := range records {
+		// Check if record type is supported
+		if !supported[rc.Type] {
+			errs = append(errs, fmt.Errorf("joker does not support %s records", rc.Type))
+			continue
+		}
+
+		// Check TTL minimum - NAPTR and SVC records can have TTL=0, others need >= 300
+		if rc.TTL != 0 && rc.TTL < 300 {
+			if rc.Type != "NAPTR" && rc.Type != "SVC" {
+				errs = append(errs, fmt.Errorf("joker requires TTL to be 300 or higher (except NAPTR and SVC which can be 0)"))
+			}
+		}
```

Comment:
Rather than reject bad TTLs, it's best to just correct them and output a warning.  See:
* providers/loopia/loopiaProvider.go
* providers/hostingde/hostingdeProvider.go
* providers/cloudns/cloudnsProvider.go

Reply:
I hope you're fine if we just skip the warning.